### PR TITLE
Fixed #28596 -- Fixed QuerySet.bulk_create() and cascade deletion crash on Oracle when using more than 65535 parameters.

### PR DESCRIPTION
**Pull request by felixxm**
https://code.djangoproject.com/ticket/28596

### CODE REVIEW

**Comment by felixxm** [comment]
buildbot, test on oracle.

**Comment by felixxm** [comment]
`delete.tests.DeletionTests.test_large_delete_related ` failure on Oracle is not related with this PR. I've added separate PR with fix #9180.

**Comment by felixxm** [comment]
buildbot, test on oracle.

**Comment by felixxm** [comment]
buildbot, test on oracle.

**Comment by felixxm** [comment]
buildbot, test on oracle.

**Comment by felixxm** [comment]
@fingon Can you confirm that this patch fixes the reported bug?

**Comment by fingon** [comment]
We are using similar monkeypatch and it fixed the problem for us.

File: django/db/backends/oracle/features.py [hunk 1]
```diff
@@ -55,3 +55,5 @@ class DatabaseFeatures(BaseDatabaseFeatures):
     """
     supports_callproc_kwargs = True
     supports_over_clause = True
+    # Oracle has limitation of 2^16 - 1 parameters in a query.
```
Comment (timgraham) [inline django/db/backends/oracle/features.py]: "Oracle is limited to 2^16 - 1 parameters in a query."
or maybe it's better to write `max_query_params = 2^16 − 1` and omit the comment?

File: django/db/backends/oracle/operations.py [hunk 2]
```diff
@@ -553,3 +553,9 @@ def subtract_temporals(self, internal_type, lhs, rhs):
             rhs_sql, rhs_params = rhs
             return "NUMTODSINTERVAL(%s - %s, 'DAY')" % (lhs_sql, rhs_sql), lhs_params + rhs_params
         return super().subtract_temporals(internal_type, lhs, rhs)
+
+    def bulk_batch_size(self, fields, objs):
+        """Oracle has limitation of number of parameters in a query."""
```
Comment (timgraham) [inline django/db/backends/oracle/operations.py]: Oracle restricts the number of parameters in a query.

File: tests/backends/oracle/test_operations.py [hunk 3]
```diff
@@ -9,3 +9,17 @@ class OperationsTests(unittest.TestCase):
     def test_sequence_name_truncation(self):
         seq_name = connection.ops._get_no_autofield_sequence_name('schema_authorwithevenlongee869')
         self.assertEqual(seq_name, 'SCHEMA_AUTHORWITHEVENLOB0B8_SQ')
+
+    def test_bulk_batch_size(self):
+        # Oracle has limitation of 2^16 - 1 parameters in a query.
```
Comment (timgraham) [inline tests/backends/oracle/test_operations.py]: same suggested comment as before

**Comment by felixxm** [comment]
@timgraham Thanks for review.

**Pull request merged**